### ISSUE TITLE
problem starting fauxmo from Raspian service

### ISSUE THREAD

**Issue by johngo7470** [n8henrie/fauxmo#82]
- Issues not following the template may be closed for that reason alone.
- **Operating system and version**: Raspian 4.19.66
- **Python version**: 3.7
- **Fauxmo version** (`fauxmo --version`): 0.4.9
- **Echo device type** (e.g. Echo, Echo Plus, Dot):  n/a
- **[Echo Firmware Version](https://github.com/n8henrie/fauxmo/blob/dev/README.md#troubleshooting--faq)**: n/a

## My Issue
Similar to another user, I'm able to start up fauxmo flawlessly, by using the following command, logged in as user 'pi':

     fauxmo -c /home/pi/config.json

pi@raspberrypi:/lib/systemd/system $ fauxmo --version
v0.4.9
pi@raspberrypi:/lib/systemd/system $ whereis fauxmo
fauxmo: /home/pi/.local/bin/fauxmo

The system file contains: ExecStart=/home/pi/.local/bin/fauxmo -c /home/pi/config.json

The problem noted in the system status, is: 

     Sep 09 21:23:52 raspberrypi fauxmo[696]:   File "/home/pi/.local/bin/fauxmo", line 6, in <module>
     Sep 09 21:23:52 raspberrypi fauxmo[696]:     from fauxmo.cli import cli
     Sep 09 21:23:52 raspberrypi fauxmo[696]: ModuleNotFoundError: No module named 'fauxmo'
     Sep 09 21:23:52 raspberrypi systemd[1]: myfauxmo.service: Main process exited, code=exited, status=1/FAILURE
     Sep 09 21:23:52 raspberrypi systemd[1]: myfauxmo.service: Failed with result 'exit-code'.

The contents of /home/pi/.local/bin/fauxmo:

     pi@raspberrypi:~/.local/bin $ cat fauxmo
     #!/usr/bin/python3.7
     # -*- coding: utf-8 -*-
     import re
     import sys

     from fauxmo.cli import cli

     if __name__ == '__main__':
         sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
         sys.exit(cli())

==================

On a related note, if fauxmo is installed in a venv (e.g. /home/pi/.venv), how can this be accommodated in a module.service startup script?  Do I need to add an activate command to the .service file?

## [WHYT](http://mattgemmell.com/what-have-you-tried)

---

Please make sure you've taken these steps before submitting a new issue:

- [ ] Try to reproduce the issue with Fauxmo installed in a
  [venv](https://docs.python.org/3/library/venv.html)
- [ ] Ensure you're running a supported version of Python
    - Requires Python >= 3.2 for Fauxmo < 0.3.0
    - Requires Python >= 3.4.4 for Fauxmo >= 0.3.0
    - Requires Python >= 3.6.0 for Fauxmo >= 0.4.0
- [ ] Run Fauxmo in debug mode (`-vvv`) and include relevant output
- [ ] Include your Fauxmo `config.json`
- [ ] Search the existing (including closed) issues
- [ ] Please use [codeblocks][1] around your debug output and config

[1]: https://help.github.com/articles/creating-and-highlighting-code-blocks/

**Comment by johngo7470** [comment]
I might have figured this one out... 

I just noticed your .service example has the .venv version in there - will try this later this evening.

However, what path should be used for the "Working Directory"? (which should probably be moved below the line "#Fix the paths below:")

   pi@raspi:/ $ sudo find . -name fauxmo 2>/dev/null
   ./home/pi/venv/bin/fauxmo
   ./home/pi/venv/lib/python3.6/site-packages/fauxmo
   ./usr/local/bin/fauxmo
   ./usr/local/lib/python3.6/site-packages/fauxmo
   ./opt/pyenv/versions/3.7.3/bin/fauxmo
   ./opt/pyenv/versions/3.7.3/lib/python3.7/site-packages/fauxmo
   ./opt/pyenv/shims/fauxmo

In my particular case, I *think* I'm using the non-venv version (still trying to understand venv, before I switch to it).

**Comment by johngo7470** [comment]
I first installed fauxmo using your instructions under "Simple install: From PyPI".

Then, being unsure of using venv, I did a "python3 -m pip remove fauxmo", followed by "python3.7 -m pip install fauxmo".


[And is this your system Python install?]

If I understand your question, yes... It's a Raspberry Pi 4, created a few weeks ago, so it would have likely been created with Python 3.7 from the get-go. When I type 'python3', it shows the version is 3.7.3, whereas on another Raspberry I'm working on (RPi 3), when I start up 'python3', it comes back with 3.5.3, even though I know 3.7.3 is available somewhere (I think via pyenv).

**Comment by n8henrie** [comment]
> I first installed fauxmo using your instructions under "Simple install: From PyPI".

I don't think this is possible, since my [simple install instructions](https://github.com/n8henrie/fauxmo#simple-install-from-pypi) install into a venv, so your `fauxmo` CLI script would not be in `~/.local/bin`. It should definitely have `.venv` somewhere in its path if you had followed those instructions verbatim.

My guess is at some point you ran into a permissions error and instead ran: `python3 -m pip install --user fauxmo`. Is that possible?

And then when you ran `pip uninstall fauxmo` (`remove` is not a valid pip command to my knowledge) perhaps you got left with an artifact in `~/.local/bin`.

Regardless, glad you got it working. venv is the way to go, although Fauxmo doesn't have any dependencies, so theoretically it would probably work fine in a system-wide installation.
